src/Common/src/TypeSystem/Common/TypeSystemContext.cs

```diff
@@ -736,5 +736,42 @@ protected internal virtual bool ComputeHasGCStaticBase(FieldDesc field)
             // Type system contexts that support this need to override this.
             throw new NotSupportedException();
         }
+
+        /// <summary>
+        /// Extension point for computation of type flags. This extension point may be overriden to customize
+        /// type flags that are not computed by the ComputeTypeFlags api on TypeDesc.
+        /// At this time, the flag so computed are:
+        /// HasStaticConstructor
+        /// 
+        /// When overriding, return the value of calling this base implementation passing the flags value computed
+        /// in the override.
+        /// </summary>
+        protected internal virtual TypeFlags ComputeTypeFlags(TypeDesc type, TypeFlags flags, TypeFlags mask)
+        {
+            // If we are looking to compute HasStaticConstructor, and we haven't yet assigned a value
+            if (((mask & TypeFlags.HasStaticConstructorComputed) == TypeFlags.HasStaticConstructorComputed) &&
+                ((flags & TypeFlags.HasStaticConstructorComputed) == 0))
```

Comment:
The idea here is that a derived TypeSystemContext could have computed the static constructor information, but would still be implemented by calling base to possibly handle other flags. In that case, the derived one would need to have logic to strip the mask down. If you'd prefer that approach, I can certainly make it so.